scipy/stats/_distribution_infrastructure.py

```diff
@@ -0,0 +1,5258 @@
+            parameters of this parameterization.
+        """
+        return parameters == set(self.parameters.keys())
+
+    def validation(self, parameter_values):
+        r""" Input validation / standardization of parameterization.
+
+        Parameters
+        ----------
+        parameter_values : dict
+            The keyword arguments passed as parameter values to the
+            distribution.
+
+        Returns
+        -------
+        all_valid : ndarray
+            Logical array indicating the elements of the broadcasted arrays
+            for which all parameter values are valid.
+        dtype : dtype
+            The common dtype of the parameter arrays. This will determine
+            the dtype of the output of distribution methods.
+        """
+        all_valid = True
+        dtypes = set()  # avoid np.result_type if there's only one type
+        for name, arr in parameter_values.items():
+            parameter = self.parameters[name]
+            arr, dtype, valid = parameter.validate(arr, parameter_values)
+            dtypes.add(dtype)
+            all_valid = all_valid & valid
+            parameter_values[name] = arr
+        dtype = arr.dtype if len(dtypes)==1 else np.result_type(*list(dtypes))
+
+        return all_valid, dtype
+
+    def __str__(self):
+        r"""Returns a string representation of the parameterization."""
+        messages = [str(param) for name, param in self.parameters.items()]
+        return ", ".join(messages)
+
+    def draw(self, sizes=None, rng=None, proportions=None, region='domain'):
+        r"""Draw random values of all parameters for use in testing.
+
+        Parameters
+        ----------
+        sizes : iterable of shape tuples
+            The size of the array to be generated for each parameter in the
+            parameterization. Note that the order of sizes is arbitary; the
+            size of the array generated for a specific parameter is not
+            controlled individually as written.
+        rng : NumPy Generator
+            The generator used to draw random values.
+        proportions : tuple
+            A tuple of four non-negative numbers that indicate the expected
+            relative proportion of elements that are within the parameter's
+            domain, are on the boundary of the parameter's domain, are outside
+            the parameter's domain, and have value NaN. For more information,
+            see the `draw` method of the _Parameter subclasses.
+        domain : str
+            The domain of the `_Parameter` from which to draw. Default is
+            "domain" (the *full* domain); alternative is "typical".
+
+        Returns
+        -------
+        parameter_values : dict (string: array)
+            A dictionary of parameter name/value pairs.
+        """
+        # ENH: be smart about the order. The domains of some parameters
+        # depend on others. If the relationshp is simple (e.g. a < b < c),
```

Comment:
I think the strict way to write this would be "The domains of some parameters may depend on the values of other parameters." It's clear enough to me what's meant here though if @mdhaber doesn't want to fix this now. I hope the natural follow-up is for someone to make the suggested enhancement.